amplify/backend/function/fetchDomainBalance/src/api/graphql/requests.js

```diff
@@ -0,0 +1,202 @@
+    colonyAddress,
+    limit,
+    nextToken
+  });
+
+  if (!result) {
+    console.warn(
+      'Could not find any colony funds claims in db.',
+    );
+  }
+
+  return result.data.getFundsClaimsByColony;
+};
+
+const getAllIncomingFunds = async (colonyAddress, domain) => {
+  if (domain && !domain.isRoot) {
+    return []
+  }
+  return getAllPages(getIncomingFundsData, { colonyAddress })
+};
+
+const getExpendituresData = async ({
+  nativeDomainId, limit, nextToken
+}) => {
+  const result = await graphqlRequest(getDomainExpenditures, {
+    nativeDomainId,
+    limit,
+    nextToken
+  });
+
+  if (!result) {
+    console.warn(
+      'Could not find any domain expenditure in db.',
+    );
+  }
+
+  return result.data.listExpenditures;
+};
+
+const getAllExpenditures = async (domain) => {
+  if (!domain?.nativeId) {
+    return []
+  }
+  return getAllPages(getExpendituresData, { nativeDomainId: domain?.nativeId })
+};
+
+const acceptedColonyActionTypes = [
+  "PAYMENT", "PAYMENT_MOTION",
+  "MOVE_FUNDS", "MOVE_FUNDS_MOTION"
+];
+
+const getActionsData = async ({
+  colonyAddress, domainAddress, nextToken, limit = 2
+}) => {
+  const result = await graphqlRequest(getColonyActions, {
+    colonyAddress,
+    filter: {
+      and: [{
+        or: [{ fromDomainId: { eq: domainAddress } }, { toDomainId: { eq: domainAddress } }]
+      }, {
+        or: acceptedColonyActionTypes.map(acceptedColonyActionType => ({ type: { eq: acceptedColonyActionType } }))
+      }, {
+        showInActionsList: { eq: true }
+      }
+      ]
+    },
+    limit,
+    nextToken
+  });
+
+  if (!result) {
+    console.warn(
+      'Could not find any colony actions in db.',
+    );
+  }
+
+  return result.data.getActionsByColony;
+}
+
+const getAllActions = async (colonyAddress, domainAddress) => {
+  return getAllPages(getActionsData, { colonyAddress, domainAddress })
+}
+
+const getTokensDecimalsFor = async (tokenAddresses) => {
+  const tokenDecimals = {};
+
+  const tokenResponses = await Promise.all(tokenAddresses.map(tokenAddress => graphqlRequest(getToken, {
+    tokenAddress
+  })))
+
+  tokenResponses.forEach(tokenResponse => {
+    const { id, decimals } = tokenResponse?.data?.getToken || {};
+    tokenDecimals[id] = decimals;
+  })
+
+  return tokenDecimals;
+};
+
+const saveExchangeRate = async ({ tokenId, date, marketPrice }) => {
+
+  const result = await graphqlRequest(saveTokenExchangeRate, {
+    input: {
+      tokenId,
+      date,
+      marketPrice,
+    }
+  });
+
+  return result.data.createTokenExchangeRate;
+
+}
+
+const getExchangeRate = async ({ tokenId, date }) => {
+
+  const result = await graphqlRequest(getTokenExchangeRate, {
+    tokenId,
+    date,
+  });
+
+  return result.data?.tokenExhangeRateByTokenId?.items?.[0];
+}
+
+module.exports = {
+  getAllIncomingFunds,
+  getAllExpenditures,
+  getAllActions,
+  getDomains,
+  saveExchangeRate,
+  getExchangeRate,
+  getTokensDecimalsFor
+};
```

Comment:
The lack of empty line made me realise we don't auto-run prettier for `.js` files - bonus points if you could add an extra rule to `.lintstagedrc` 😄